### DISCUSSION TITLE
Plugin author best practices for adding UI?

### DISCUSSION THREAD

**Discussion by SUPERCILEX** [bevyengine/bevy#3770] [Q&A] [answered]
What's the best practice for:
- Fonts. Should plugins use something like `include_bytes` to force the font to be there? That'll cause binary bloat. So should some documented name be picked and then users can symlink their fonts into the expected file for the plugin to load?
- Cameras. Should UI plugins add the `UiCameraBundle`? What happens if the end user does that too?

**Comment by alice-i-cecile** [comment, marked as answer]
For fonts: ideally, we should be loading system fonts, especially for defaults. But that's not in place yet, see #1017. For now, I'd probably have users pass in a path during plugin creation.

For camera: I would probably not spawn a UI camera, and note it in your docs. But see #3679, which will solve this for you.

**Reply by SUPERCILEX** [reply]
Thank you! It's awesome to see how quickly things are improving.
